Widgets/RoomView.cpp

```diff
@@ -22,49 +22,55 @@ bool InstanceSortFilterProxyModel::lessThan(const QModelIndex &left, const QMode
   return objA->data(Object::kDepthFieldNumber) < objB->data(Object::kDepthFieldNumber);
 }
 
-RoomView::RoomView(QWidget* parent) : AssetView(parent), model(nullptr) {
-  this->SetZoom(1.0);
+RoomView::RoomView(AssetScrollAreaBackground* parent) : AssetView(parent), model(nullptr) {
+  setFixedSize(sizeHint());
   this->sortedInstances = new InstanceSortFilterProxyModel(this);
   this->sortedTiles = new QSortFilterProxyModel(this);
 }
 
 void RoomView::SetResourceModel(ProtoModelPtr model) {
   this->model = model;
-  this->sortedInstances->setSourceModel(model->GetRepeatedSubModel(Room::kInstancesFieldNumber));
-  this->sortedInstances->sort(Room::Instance::kObjectTypeFieldNumber);
-  this->sortedTiles->setSourceModel(model->GetRepeatedSubModel(Room::kTilesFieldNumber));
-  this->sortedTiles->sort(Room::Tile::kDepthFieldNumber);
-  this->SetZoom(1.0);
+
+  if (model != nullptr) {
```

Comment:
Out of curiosity, what prompted you to add this check?

Reply:
When you unset the room in the pathview you set it to nullptr so without this check it'll crash